### DISCUSSION TITLE
Is it safe to remove tag CSS classes from HTML layouts?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#799] [Ask a question] [answered]
I'm posting this as a general question first, please let me know if the following should be investigated further as a potential bug. 🙂

Background: For my [Wittstock theme](https://github.com/schulle4u/yellow-wittstock) the goal was to remove all `<div>` elements and unnecessary classes from the layout files and replace them with classless semantic HTML5 elements where possible. However some `<div>` elements or even their classes seem to have an effect on how Yellow works, primarily the edit extension. One such example is the `<div>` below the body tag which wraps the whole page output. Parts of the edit extension don't work as expected if I remove it in my layout, thats comprehensible so far. 

Another one which causes very special and odd behaviour in my theme is the entry class wrapper for blog and wiki pages. It fetches the page tags and uses them as class names in the opening `<div>` element (e. g `<div class="entry tag-example tag-example2">`). I've left the entry class, but replaced the wrapper with an `<article>` tag. If a blog page only has one tag, everything looks OK. If more than one tag is present, and I open the blog page in the firefox reading view (F9), the page somehow gets messed up and displays navigation elements which shouldn't be there. Only in firefox, and only in my theme. I wasn't able to track down the problem, the markup validation doesn't show related issues, and there are no php errors or unusual log messages. The simplest solution for the moment would be to just remove these tag entry classes. Is there a technical reason for these class names or is it safe to remove them?

**Comment by markseuffert** [comment, marked as answer]
Yes, you can remove these CSS classes from your own HTML layouts. 

They are intended for web designers and will transfer some of the page metadata into the HTML code. In the [wiki extension](https://github.com/annaesvensson/yellow-wiki) and [blog extension](https://github.com/annaesvensson/yellow-blog) generated HTML pages from layouts and generated HTML fragments from shortcuts contain these CSS classes. For example you could visually highlight new wiki pages with the `Tag: new` or  visually highlight popular blog pages with the `Tag: popular`. It's up to the web designer to use this metadata for something useful.

**Reply by schulle4u** [reply]
Thanks, thats what I've already supposed. But since class names also can be used by JS modules I wanted to check twice.
